extension/src/popup/components/sendPayment/SendConfirm/TransactionDetails/index.tsx

```diff
@@ -1,120 +1,205 @@
+  const transactionHash = submission.response?.hash;
+  const publicKey = useSelector(publicKeySelector);
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const [assetIcons, setAssetIcons] = useState({} as AssetIcons);
 
-    let horizonAsset: Asset;
-    if (asset === "native") {
-      horizonAsset = StellarSdk.Asset.native();
-    } else {
-      horizonAsset = new StellarSdk.Asset(
-        asset.split(":")[0],
-        asset.split(":")[1],
-      );
-    }
+  let horizonAsset = StellarSdk.Asset.native();
+  if (asset.includes(":")) {
+    horizonAsset = new StellarSdk.Asset(
+      asset.split(":")[0],
+      asset.split(":")[1],
+    );
+  }
+  const assetTotals = [
+    {
+      token: { issuer: horizonAsset.issuer, code: horizonAsset.code },
+      total: amount || "0",
+    },
+  ];
 
-    const transactionXDR = await server
-      .loadAccount(publicKey)
-      .then((sourceAccount: Types.Account) => {
-        const transaction = new StellarSdk.TransactionBuilder(sourceAccount, {
-          fee: transactionFee,
-          networkPassphrase: networkDetails.networkPassphrase,
-        })
-          .addOperation(
-            StellarSdk.Operation.payment({
-              destination,
-              asset: horizonAsset,
-              amount,
-            }),
-          )
-          .addMemo(StellarSdk.Memo.text(memo))
-          .setTimeout(180)
-          .build();
-
-        return transaction.toXDR();
+  useEffect(() => {
+    (async () => {
+      const iconURL = await getIconUrlFromIssuer({
+        key: horizonAsset.issuer,
+        code: horizonAsset.code,
+        networkDetails,
       });
+      setAssetIcons({ [horizonAsset.code]: iconURL });
+    })();
+  }, [horizonAsset.code, horizonAsset.issuer, networkDetails]);
 
-    const res = await dispatch(
-      signFreighterTransaction({
-        transactionXDR,
-        network: networkDetails.networkPassphrase,
-      }),
-    );
+  // handles signing and submitting
+  const handleSend = async () => {
+    const server = new StellarSdk.Server(networkDetails.networkUrl);
 
-    if (signFreighterTransaction.fulfilled.match(res)) {
-      const signedXDR = StellarSdk.TransactionBuilder.fromXDR(
-        res.payload.signedTransaction,
-        networkDetails.networkPassphrase,
-      );
+    try {
+      const transactionXDR = await server
+        .loadAccount(publicKey)
+        .then((sourceAccount: Types.Account) => {
+          const transaction = new StellarSdk.TransactionBuilder(sourceAccount, {
+            fee: xlmToStroop(transactionFee).toString(),
+            networkPassphrase: networkDetails.networkPassphrase,
+          })
+            .addOperation(
+              StellarSdk.Operation.payment({
+                destination,
+                asset: horizonAsset,
+                amount,
+              }),
+            )
+            .addMemo(StellarSdk.Memo.text(memo))
+            .setTimeout(180)
```

Comment:
jc - what does setting a timeout of 180 do here?

Reply:
setting the [timebounds](https://developers.stellar.org/docs/glossary/transactions/#time-bounds) for the transaction. (I noticed we aren't using it in all repos atm but according to our docs it's good practice). 180 is what we use in our example docs so I went with that, not opposed to changing it though. wdyt?

Reply:
Oh cool - I hadn't caught that in the docs before. I wonder if `180` is the standard and if we should set that as a const in the codebase 🤔 Not something we need to solve now!